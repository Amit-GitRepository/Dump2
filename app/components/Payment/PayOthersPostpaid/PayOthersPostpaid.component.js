import Proptypes from 'prop-types';
import React, {Component} from 'react';
import typeMap from '../../../config/productTypeMap.config';
import WrapperCard from '../../Shared/WrapperCard/WrapperCard.component';
import {AmountText, Button, ISText} from '../../Shared';
import {noop, result} from 'lodash';
import {phoneNumberFormatter} from '../../../utils/common.util';
import {ScrollView, View} from 'react-native';
import {styles} from './PayOthersPostpaid.style';
import {TMPostpaidProductCard, TOnlineProductCard, TVisionProductCard} from '../../BillUsage/BillUsage/ProductCards';
import {translate} from '../../../language/i18n/helper';

class PayOthersPostpaid extends Component {
  getProductCard = () => {
    const {product, dueBills, toggleProductCheck, toggleProductCollapse, onBillDetailsClick, toggleBillDetailCheck} = this.props;
    const {productId, productType, accountId, isCollapsed, balance, statusCode, isChecked} = product;
    const phoneNumber = phoneNumberFormatter(productId);
    const dueBillsForProduct = result(dueBills, `${accountId}`);
    const oncheckToggle = () => toggleProductCheck();
    const onCollapseToggle = () => toggleProductCollapse(!isCollapsed, accountId);
    switch (productType) {
    case 'TrueMoveH' : {
      return (<TMPostpaidProductCard phone={phoneNumber} amount={balance} statusCode={statusCode} dueBills={dueBillsForProduct} isCollapsed={isCollapsed}
        onCollapseToggle={onCollapseToggle} onChecked={oncheckToggle} isChecked={isChecked} onBillDetailsClick={onBillDetailsClick} toggleBillDetailCheck={toggleBillDetailCheck(accountId)} isOthers={true}/>);
    }
    case 'TrueOnline': {
      return (<TOnlineProductCard phone={phoneNumber} amount={balance} statusCode={statusCode} dueBills={dueBillsForProduct} isCollapsed={isCollapsed}
        onCollapseToggle={onCollapseToggle} onChecked={oncheckToggle} isChecked={isChecked} onBillDetailsClick={onBillDetailsClick} toggleBillDetailCheck={toggleBillDetailCheck(accountId)} isOthers={true}/>);
    }
    case 'TrueVision': {
      return (<TVisionProductCard phone={phoneNumber} amount={balance} statusCode={statusCode} dueBills={dueBillsForProduct} isCollapsed={isCollapsed}
        onCollapseToggle={onCollapseToggle} onChecked={oncheckToggle} isChecked={isChecked} onBillDetailsClick={onBillDetailsClick} toggleBillDetailCheck={toggleBillDetailCheck(accountId)} isOthers={true}/>);
    }
    }
  }

  navigateToPayments = () => {
    const {product: {productType, accountId, isChecked, checkedBillSum}, goToScreen, setPaymentItems} = this.props;
    const accountNumber = accountId;
    if (isChecked) {
      const productToPay = [{
        service: typeMap[productType],
        amount: checkedBillSum,
        accountNumber
      }];
      setPaymentItems(productToPay);
      goToScreen('Payment');
    }
  }

  render () {
    const {product} = this.props;
    const totalAmount = product.isChecked ? product.checkedBillSum : 0;
    return (
      <ScrollView contentContainerStyle={styles.container}>
        <WrapperCard style={styles.wrapperCard} contentContainerStyles={styles.contentContainer}>
          <View style={styles.totalBillSection}>
            <View>
              <ISText style={styles.postpaidTitleText}>{translate('BILLS_USAGE_TOTAL')}</ISText>
              <AmountText style={styles.totalAmount} value={totalAmount} isTotalText/>
            </View>
            <Button text={translate('BILLS_USAGE_PAY')} disabled={!totalAmount} textStyle={styles.buttonText} style={styles.buttonStyle} onPress={this.navigateToPayments}/>
          </View>
          {this.getProductCard()}
        </WrapperCard>
      </ScrollView>
    );
  }
}
PayOthersPostpaid.defaultProps = {
  dueBills: {},
  product: {},
  toggleProductCheck: noop,
  goToScreen: noop,
  setPaymentItems: noop
};
PayOthersPostpaid.propTypes = {
  product: Proptypes.object.isRequired,
  dueBills: Proptypes.object,
  toggleProductCheck: Proptypes.func,
  toggleProductCollapse: Proptypes.func,
  toggleBillDetailCheck: Proptypes.func,
  onBillDetailsClick: Proptypes.func,
  setPaymentItems: Proptypes.func,
  goToScreen: Proptypes.func
};
export default PayOthersPostpaid;
