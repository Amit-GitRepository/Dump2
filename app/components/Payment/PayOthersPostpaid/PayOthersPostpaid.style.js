import {colors, fonts} from '../../../themes/constants.styles';

export const styles = {
  scrollContainer: {
    flex: 1,
    alignItems: 'stretch',
    backgroundColor: colors.GRAY_NURSE,
    marginBottom: 20
  },
  container: {
    padding: 10
  },
  buttonText: {
    fontSize: fonts.FONT_SIZE_MEDIUM
  },
  buttonStyle: {
    paddingHorizontal: 25,
    paddingVertical: 8,
    alignSelf: 'center',
    borderRadius: 30
  },
  wrapperCard: {
    flex: 0
  },
  totalAmount: {
    padding: 0,
    marginTop: -10
  },
  postpaidTitleText: {
    color: colors.PRIMARY_TEXT_TAB_LABEL,
    fontSize: fonts.FONT_SIZE_NORMAL
  },
  contentContainer: {
    paddingHorizontal: 10,
    paddingBottom: 10
  },
  totalBillSection: {
    paddingTop: 20,
    paddingLeft: 5,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between'
  }
};