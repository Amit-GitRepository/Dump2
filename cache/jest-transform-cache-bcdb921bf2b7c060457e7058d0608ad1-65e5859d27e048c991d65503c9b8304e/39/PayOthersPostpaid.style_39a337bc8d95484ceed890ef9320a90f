Object.defineProperty(exports,"__esModule",{value:true});exports.styles=undefined;var cov_22j7xuj3fb=function(){var path='/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/Payment/PayOthersPostpaid/PayOthersPostpaid.style.js',hash='19309848de6d9e7119ad41a1b00fd16d3b3542f5',global=new Function('return this')(),gcv='__coverage__',coverageData={path:'/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/Payment/PayOthersPostpaid/PayOthersPostpaid.style.js',statementMap:{'0':{start:{line:3,column:22},end:{line:44,column:1}}},fnMap:{},branchMap:{},s:{'0':0},f:{},b:{},_coverageSchema:'332fd63041d2c1bcb487cc26dd0d5f7d97098a6c'},coverage=global[gcv]||(global[gcv]={});if(coverage[path]&&coverage[path].hash===hash){return coverage[path];}coverageData.hash=hash;return coverage[path]=coverageData;}();var _constants=require('../../../themes/constants.styles');

var styles=exports.styles=(cov_22j7xuj3fb.s[0]++,{
scrollContainer:{
flex:1,
alignItems:'stretch',
backgroundColor:_constants.colors.GRAY_NURSE,
marginBottom:20},

container:{
padding:10},

buttonText:{
fontSize:_constants.fonts.FONT_SIZE_MEDIUM},

buttonStyle:{
paddingHorizontal:25,
paddingVertical:8,
alignSelf:'center',
borderRadius:30},

wrapperCard:{
flex:0},

totalAmount:{
padding:0,
marginTop:-10},

postpaidTitleText:{
color:_constants.colors.PRIMARY_TEXT_TAB_LABEL,
fontSize:_constants.fonts.FONT_SIZE_NORMAL},

contentContainer:{
paddingHorizontal:10,
paddingBottom:10},

totalBillSection:{
paddingTop:20,
paddingLeft:5,
flexDirection:'row',
alignItems:'center',
justifyContent:'space-between'}});