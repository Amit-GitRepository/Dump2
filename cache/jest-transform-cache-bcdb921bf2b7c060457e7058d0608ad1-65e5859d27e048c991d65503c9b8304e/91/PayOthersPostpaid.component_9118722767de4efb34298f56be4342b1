Object.defineProperty(exports,"__esModule",{value:true});var cov_2j8ruc13jx=function(){var path='/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/Payment/PayOthersPostpaid/PayOthersPostpaid.component.js',hash='410ebc3ba760dc9e5bae950bd21bdaf1b1bb1d57',global=new Function('return this')(),gcv='__coverage__',coverageData={path:'/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/Payment/PayOthersPostpaid/PayOthersPostpaid.component.js',statementMap:{'0':{start:{line:15,column:118},end:{line:15,column:128}},'1':{start:{line:16,column:93},end:{line:16,column:100}},'2':{start:{line:17,column:24},end:{line:17,column:55}},'3':{start:{line:18,column:31},end:{line:18,column:63}},'4':{start:{line:19,column:26},end:{line:19,column:52}},'5':{start:{line:19,column:32},end:{line:19,column:52}},'6':{start:{line:20,column:29},end:{line:20,column:81}},'7':{start:{line:20,column:35},end:{line:20,column:81}},'8':{start:{line:21,column:4},end:{line:34,column:5}},'9':{start:{line:23,column:6},end:{line:24,column:208}},'10':{start:{line:27,column:6},end:{line:28,column:208}},'11':{start:{line:31,column:6},end:{line:32,column:208}},'12':{start:{line:38,column:104},end:{line:38,column:114}},'13':{start:{line:39,column:26},end:{line:39,column:35}},'14':{start:{line:40,column:4},end:{line:48,column:5}},'15':{start:{line:41,column:27},end:{line:45,column:8}},'16':{start:{line:46,column:6},end:{line:46,column:36}},'17':{start:{line:47,column:6},end:{line:47,column:28}},'18':{start:{line:52,column:22},end:{line:52,column:32}},'19':{start:{line:53,column:24},end:{line:53,column:70}},'20':{start:{line:54,column:4},end:{line:67,column:6}},'21':{start:{line:70,column:0},end:{line:76,column:2}},'22':{start:{line:77,column:0},end:{line:86,column:2}}},fnMap:{'0':{name:'(anonymous_0)',decl:{start:{line:14,column:19},end:{line:14,column:20}},loc:{start:{line:14,column:25},end:{line:35,column:3}},line:14},'1':{name:'(anonymous_1)',decl:{start:{line:19,column:26},end:{line:19,column:27}},loc:{start:{line:19,column:32},end:{line:19,column:52}},line:19},'2':{name:'(anonymous_2)',decl:{start:{line:20,column:29},end:{line:20,column:30}},loc:{start:{line:20,column:35},end:{line:20,column:81}},line:20},'3':{name:'(anonymous_3)',decl:{start:{line:37,column:23},end:{line:37,column:24}},loc:{start:{line:37,column:29},end:{line:49,column:3}},line:37},'4':{name:'(anonymous_4)',decl:{start:{line:51,column:2},end:{line:51,column:3}},loc:{start:{line:51,column:12},end:{line:68,column:3}},line:51}},branchMap:{'0':{loc:{start:{line:21,column:4},end:{line:34,column:5}},type:'switch',locations:[{start:{line:22,column:4},end:{line:25,column:5}},{start:{line:26,column:4},end:{line:29,column:5}},{start:{line:30,column:4},end:{line:33,column:5}}],line:21},'1':{loc:{start:{line:40,column:4},end:{line:48,column:5}},type:'if',locations:[{start:{line:40,column:4},end:{line:48,column:5}},{start:{line:40,column:4},end:{line:48,column:5}}],line:40},'2':{loc:{start:{line:53,column:24},end:{line:53,column:70}},type:'cond-expr',locations:[{start:{line:53,column:44},end:{line:53,column:66}},{start:{line:53,column:69},end:{line:53,column:70}}],line:53}},s:{'0':0,'1':0,'2':0,'3':0,'4':0,'5':0,'6':0,'7':0,'8':0,'9':0,'10':0,'11':0,'12':0,'13':0,'14':0,'15':0,'16':0,'17':0,'18':0,'19':0,'20':0,'21':0,'22':0},f:{'0':0,'1':0,'2':0,'3':0,'4':0},b:{'0':[0,0,0],'1':[0,0],'2':[0,0]},_coverageSchema:'332fd63041d2c1bcb487cc26dd0d5f7d97098a6c'},coverage=global[gcv]||(global[gcv]={});if(coverage[path]&&coverage[path].hash===hash){return coverage[path];}coverageData.hash=hash;return coverage[path]=coverageData;}();var _createClass=function(){function defineProperties(target,props){for(var i=0;i<props.length;i++){var descriptor=props[i];descriptor.enumerable=descriptor.enumerable||false;descriptor.configurable=true;if("value"in descriptor)descriptor.writable=true;Object.defineProperty(target,descriptor.key,descriptor);}}return function(Constructor,protoProps,staticProps){if(protoProps)defineProperties(Constructor.prototype,protoProps);if(staticProps)defineProperties(Constructor,staticProps);return Constructor;};}();var _propTypes=require('prop-types');var _propTypes2=_interopRequireDefault(_propTypes);
var _react=require('react');var _react2=_interopRequireDefault(_react);
var _productTypeMap=require('../../../config/productTypeMap.config');var _productTypeMap2=_interopRequireDefault(_productTypeMap);
var _WrapperCard=require('../../Shared/WrapperCard/WrapperCard.component');var _WrapperCard2=_interopRequireDefault(_WrapperCard);
var _Shared=require('../../Shared');
var _lodash=require('lodash');
var _common=require('../../../utils/common.util');
var _reactNative=require('react-native');
var _PayOthersPostpaid=require('./PayOthersPostpaid.style');
var _ProductCards=require('../../BillUsage/BillUsage/ProductCards');
var _helper=require('../../../language/i18n/helper');function _interopRequireDefault(obj){return obj&&obj.__esModule?obj:{default:obj};}function _classCallCheck(instance,Constructor){if(!(instance instanceof Constructor)){throw new TypeError("Cannot call a class as a function");}}function _possibleConstructorReturn(self,call){if(!self){throw new ReferenceError("this hasn't been initialised - super() hasn't been called");}return call&&(typeof call==="object"||typeof call==="function")?call:self;}function _inherits(subClass,superClass){if(typeof superClass!=="function"&&superClass!==null){throw new TypeError("Super expression must either be null or a function, not "+typeof superClass);}subClass.prototype=Object.create(superClass&&superClass.prototype,{constructor:{value:subClass,enumerable:false,writable:true,configurable:true}});if(superClass)Object.setPrototypeOf?Object.setPrototypeOf(subClass,superClass):subClass.__proto__=superClass;}var

PayOthersPostpaid=function(_Component){_inherits(PayOthersPostpaid,_Component);function PayOthersPostpaid(){var _ref;var _temp,_this,_ret;_classCallCheck(this,PayOthersPostpaid);for(var _len=arguments.length,args=Array(_len),_key=0;_key<_len;_key++){args[_key]=arguments[_key];}return _ret=(_temp=(_this=_possibleConstructorReturn(this,(_ref=PayOthersPostpaid.__proto__||Object.getPrototypeOf(PayOthersPostpaid)).call.apply(_ref,[this].concat(args))),_this),_this.
getProductCard=function(){cov_2j8ruc13jx.f[0]++;var _ref2=(cov_2j8ruc13jx.s[0]++,
_this.props),product=_ref2.product,dueBills=_ref2.dueBills,toggleProductCheck=_ref2.toggleProductCheck,toggleProductCollapse=_ref2.toggleProductCollapse,onBillDetailsClick=_ref2.onBillDetailsClick,toggleBillDetailCheck=_ref2.toggleBillDetailCheck;var _ref3=(cov_2j8ruc13jx.s[1]++,
product),productId=_ref3.productId,productType=_ref3.productType,accountId=_ref3.accountId,isCollapsed=_ref3.isCollapsed,balance=_ref3.balance,statusCode=_ref3.statusCode,isChecked=_ref3.isChecked;
var phoneNumber=(cov_2j8ruc13jx.s[2]++,(0,_common.phoneNumberFormatter)(productId));
var dueBillsForProduct=(cov_2j8ruc13jx.s[3]++,(0,_lodash.result)(dueBills,''+accountId));cov_2j8ruc13jx.s[4]++;
var oncheckToggle=function oncheckToggle(){cov_2j8ruc13jx.f[1]++;cov_2j8ruc13jx.s[5]++;return toggleProductCheck();};cov_2j8ruc13jx.s[6]++;
var onCollapseToggle=function onCollapseToggle(){cov_2j8ruc13jx.f[2]++;cov_2j8ruc13jx.s[7]++;return toggleProductCollapse(!isCollapsed,accountId);};cov_2j8ruc13jx.s[8]++;
switch(productType){
case'TrueMoveH':cov_2j8ruc13jx.b[0][0]++;{cov_2j8ruc13jx.s[9]++;
return _react2.default.createElement(_ProductCards.TMPostpaidProductCard,{phone:phoneNumber,amount:balance,statusCode:statusCode,dueBills:dueBillsForProduct,isCollapsed:isCollapsed,
onCollapseToggle:onCollapseToggle,onChecked:oncheckToggle,isChecked:isChecked,onBillDetailsClick:onBillDetailsClick,toggleBillDetailCheck:toggleBillDetailCheck(accountId),isOthers:true});
}
case'TrueOnline':cov_2j8ruc13jx.b[0][1]++;{cov_2j8ruc13jx.s[10]++;
return _react2.default.createElement(_ProductCards.TOnlineProductCard,{phone:phoneNumber,amount:balance,statusCode:statusCode,dueBills:dueBillsForProduct,isCollapsed:isCollapsed,
onCollapseToggle:onCollapseToggle,onChecked:oncheckToggle,isChecked:isChecked,onBillDetailsClick:onBillDetailsClick,toggleBillDetailCheck:toggleBillDetailCheck(accountId),isOthers:true});
}
case'TrueVision':cov_2j8ruc13jx.b[0][2]++;{cov_2j8ruc13jx.s[11]++;
return _react2.default.createElement(_ProductCards.TVisionProductCard,{phone:phoneNumber,amount:balance,statusCode:statusCode,dueBills:dueBillsForProduct,isCollapsed:isCollapsed,
onCollapseToggle:onCollapseToggle,onChecked:oncheckToggle,isChecked:isChecked,onBillDetailsClick:onBillDetailsClick,toggleBillDetailCheck:toggleBillDetailCheck(accountId),isOthers:true});
}}

},_this.

navigateToPayments=function(){cov_2j8ruc13jx.f[3]++;var _ref4=(cov_2j8ruc13jx.s[12]++,
_this.props),_ref4$product=_ref4.product,productType=_ref4$product.productType,accountId=_ref4$product.accountId,isChecked=_ref4$product.isChecked,checkedBillSum=_ref4$product.checkedBillSum,goToScreen=_ref4.goToScreen,setPaymentItems=_ref4.setPaymentItems;
var accountNumber=(cov_2j8ruc13jx.s[13]++,accountId);cov_2j8ruc13jx.s[14]++;
if(isChecked){cov_2j8ruc13jx.b[1][0]++;
var productToPay=(cov_2j8ruc13jx.s[15]++,[{
service:_productTypeMap2.default[productType],
amount:checkedBillSum,
accountNumber:accountNumber}]);cov_2j8ruc13jx.s[16]++;

setPaymentItems(productToPay);cov_2j8ruc13jx.s[17]++;
goToScreen('Payment');
}else{cov_2j8ruc13jx.b[1][1]++;}
},_temp),_possibleConstructorReturn(_this,_ret);}_createClass(PayOthersPostpaid,[{key:'render',value:function render()

{cov_2j8ruc13jx.f[4]++;var _ref5=(cov_2j8ruc13jx.s[18]++,
this.props),product=_ref5.product;
var totalAmount=(cov_2j8ruc13jx.s[19]++,product.isChecked?(cov_2j8ruc13jx.b[2][0]++,product.checkedBillSum):(cov_2j8ruc13jx.b[2][1]++,0));cov_2j8ruc13jx.s[20]++;
return(
_react2.default.createElement(_reactNative.ScrollView,{contentContainerStyle:_PayOthersPostpaid.styles.container},
_react2.default.createElement(_WrapperCard2.default,{style:_PayOthersPostpaid.styles.wrapperCard,contentContainerStyles:_PayOthersPostpaid.styles.contentContainer},
_react2.default.createElement(_reactNative.View,{style:_PayOthersPostpaid.styles.totalBillSection},
_react2.default.createElement(_reactNative.View,null,
_react2.default.createElement(_Shared.ISText,{style:_PayOthersPostpaid.styles.postpaidTitleText},(0,_helper.translate)('BILLS_USAGE_TOTAL')),
_react2.default.createElement(_Shared.AmountText,{style:_PayOthersPostpaid.styles.totalAmount,value:totalAmount,isTotalText:true})),

_react2.default.createElement(_Shared.Button,{text:(0,_helper.translate)('BILLS_USAGE_PAY'),disabled:!totalAmount,textStyle:_PayOthersPostpaid.styles.buttonText,style:_PayOthersPostpaid.styles.buttonStyle,onPress:this.navigateToPayments})),

this.getProductCard())));



}}]);return PayOthersPostpaid;}(_react.Component);cov_2j8ruc13jx.s[21]++;

PayOthersPostpaid.defaultProps={
dueBills:{},
product:{},
toggleProductCheck:_lodash.noop,
goToScreen:_lodash.noop,
setPaymentItems:_lodash.noop};cov_2j8ruc13jx.s[22]++;

PayOthersPostpaid.propTypes={
product:_propTypes2.default.object.isRequired,
dueBills:_propTypes2.default.object,
toggleProductCheck:_propTypes2.default.func,
toggleProductCollapse:_propTypes2.default.func,
toggleBillDetailCheck:_propTypes2.default.func,
onBillDetailsClick:_propTypes2.default.func,
setPaymentItems:_propTypes2.default.func,
goToScreen:_propTypes2.default.func};exports.default=

PayOthersPostpaid;