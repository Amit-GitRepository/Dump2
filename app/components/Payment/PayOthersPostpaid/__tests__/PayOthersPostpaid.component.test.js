import PayOthersPostpaid from '../PayOthersPostpaid.component';
import React from 'react';
import renderer from 'react-test-renderer';

describe('PayOthersPostpaid component', () => {
  it('renders correctly', () => {
    const mockProduct = {
      productId: '1234567890'
    };
    const component = renderer.create(<PayOthersPostpaid product={mockProduct} serviceNo='1234567890'/>);
    expect(component).toMatchSnapshot();
  });
});
