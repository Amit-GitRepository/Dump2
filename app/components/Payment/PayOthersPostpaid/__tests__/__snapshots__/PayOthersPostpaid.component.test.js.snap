// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PayOthersPostpaid component renders correctly 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "padding": 10,
    }
  }
>
  <View>
    <View
      style={
        Array [
          Object {
            "marginHorizontal": 8,
          },
          Object {
            "flex": 0,
          },
        ]
      }
    >
      <View
        elevation={2}
        style={
          Array [
            Object {
              "backgroundColor": "#FFFFFF",
              "borderColor": "#B9C1C9",
              "borderRadius": 10,
              "borderWidth": 1,
              "shadowColor": "#303C46",
              "shadowOffset": Object {
                "height": 2,
                "width": 1,
              },
              "shadowOpacity": 0.3,
              "shadowRadius": 1,
            },
            Object {
              "paddingBottom": 10,
              "paddingHorizontal": 10,
            },
          ]
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "paddingLeft": 5,
              "paddingTop": 20,
            }
          }
        >
          <View>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "backgroundColor": "transparent",
                  "color": "#303C46",
                  "fontFamily": "SukhumvitSet-Text",
                  "fontSize": 16,
                }
              }
            >
              Total
            </Text>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "flex-end",
                    "flexDirection": "row",
                  },
                  Object {
                    "marginTop": -10,
                    "padding": 0,
                  },
                ]
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "color": "#303C46",
                    },
                    Array [
                      Object {
                        "fontSize": 18,
                        "paddingBottom": 6,
                        "paddingRight": 3,
                      },
                      Object {},
                    ],
                    Object {
                      "fontFamily": "SukhumvitSet-Bold",
                    },
                  ]
                }
              >
                à¸¿
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "color": "#303C46",
                    },
                    Array [
                      Object {
                        "fontSize": 30,
                      },
                      Object {},
                    ],
                    Object {
                      "fontFamily": "SukhumvitSet-Bold",
                    },
                  ]
                }
              >
                0
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "color": "#303C46",
                    },
                    Array [
                      Object {
                        "color": "#68747E",
                        "fontSize": 18,
                        "paddingBottom": 5,
                      },
                      Object {},
                    ],
                    Object {
                      "fontFamily": "SukhumvitSet-Bold",
                    },
                  ]
                }
              >
                .
                00
              </Text>
            </View>
          </View>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            collapsable={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignSelf": "center",
                "backgroundColor": "#F93E60",
                "borderRadius": 30,
                "opacity": 1,
                "padding": 10,
                "paddingHorizontal": 25,
                "paddingVertical": 8,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <View
              style={
                Object {
                  "opacity": 0.7,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "alignSelf": "stretch",
                    "flexDirection": "row",
                    "justifyContent": "center",
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                        "color": "#303C46",
                      },
                      Array [
                        Object {
                          "color": "#FFFFFF",
                          "fontSize": 16,
                        },
                        Object {
                          "fontSize": 18,
                        },
                      ],
                      Object {
                        "fontFamily": "SukhumvitSet-Bold",
                      },
                    ]
                  }
                >
                  Pay
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
